bindings/python/mata.pyx

```diff
@@ -1150,3 +1464,55 @@ def run_safely_external_command(cmd: str, check_results=True, quiet=True, timeou
                 )
 
     return cmdout, cmderr
+
+cdef class Segmentation:
+    """Wrapper over Segmentation."""
+    cdef mata.CSegmentation* thisptr
+
+    def __cinit__(self, Nfa aut, Symbol symbol):
+        """
+        Compute segmentation.
+
+        :param aut: Segment automaton to compute epsilon depths for.
+        :param symbol: Symbol to execute segmentation for.
+        """
+        self.thisptr = new mata.CSegmentation(dereference(aut.thisptr), symbol)
+
+    def __dealloc__(self):
+        del self.thisptr
+
+    def get_epsilon_depths(self):
+        """
+        Get segmentation depths for ε-transitions.
+
+        :return: Map of depths to lists of ε-transitions.
+        """
+        cdef umap[unsigned, vector[CTrans]] c_epsilon_transitions = self.thisptr.get_epsilon_depths()
+        result = {}
+        for epsilon_depth_pair in c_epsilon_transitions:
```

Comment:
I'm quite surprised that this works, since I had issues with compiling the for loop of unordered maps and had to use while loop together with manual definitions of iterators. I'll do some refactoring in near future and will give the for loop chance again :>. Can you please list your version of Cython (maybe I have some older version).

Reply:
I have seen the iterators and was similarly surprised the simple for loop worked for me.

```
Cython version 0.29.32
```

Reply:
Great, thanks, I have `0.29.14`, so this could be the case. I'll try it myself and will update the requirements then.